### PR TITLE
Add file type information to the Django models

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR adds information about the file type to the Django models.

Relates to https://github.com/WordPress/openverse/discussions/85.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_api/catalog/api/models/mixins.py [hunk 1]
```diff
@@ -64,7 +66,35 @@ class FileMixin(models.Model):
     url = models.URLField(
         unique=True, max_length=1000, help_text="The actual URL to the media file."
     )
-    filesize = models.IntegerField(blank=True, null=True)
+    filesize = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text="Number in bytes, e.g. 1024.",
+        # Bytes for parity with the HTTP Content-Length header
```
Comment (zackkrida) [inline openverse_api/catalog/api/models/mixins.py:73]: 👍

File: openverse_api/catalog/api/models/image.py [hunk 2]
```diff
@@ -6,14 +6,31 @@
...
+    @property
+    def resolution_in_mp(self):  # ~ MP or megapixels
+        return (self.width * self.height) / 1e6
+
+    class Meta:
+        abstract = True
+
+
+class Image(ImageFileMixin, AbstractMedia):
+    pass
```
Comment (AetherUnbound) [inline openverse_api/catalog/api/models/image.py:32]: Neat!

**Review by AetherUnbound — APPROVED** [review]
Looks great! Are there any tests that should be updated along with this?

**Comment by dhruvkb** [comment]
@AetherUnbound I suppose the integration tests will need changing when the serializers & endpoints have updated to reflect these changes. I'm changing those in a separate PR for ease of review.

File: openverse_api/catalog/api/models/image.py (hunk 2 above)
Comment (dhruvkb) [reply openverse_api/catalog/api/models/image.py:32]: Once we abstract the file-specific outside the `Image` model, it becomes empty! Maybe we can add `alt_files` for images in the future.

**Review by obulat — APPROVED** [review]
It's nice to see all properties for alternative audio files so well-documented!

**Pull request merged**